### PR TITLE
Added mocha tests for node shim

### PR DESCRIPTION
**Pull request by mamaso**
Let's see how appveyor feels..

### CODE REVIEW

File: WebJobs.Script.proj [hunk 1]
```diff
@@ -168,6 +168,11 @@
     <xunit Assemblies="@(UnitTestAssemblies)" ExcludeTraits="$(ExcludedTestTraits)" />
   </Target>
 
+  <Target Name="MochaTest" AfterTargets="TestBuild">
+    <Exec Command="npm i mocha chai mocha-appveyor-reporter" />
```
Comment (mathewc) [inline WebJobs.Script.proj]: It'd be good to get the mocha tests to run serially with the xunit tests. That can be done as a follow up.

File: test/WebJobs.Script.Tests/TestScripts/Node/functions.tests.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+var config = process.argv[process.argv.length - 1]
+config = (config.indexOf('--config=') !== -1)? config.substr(9): 'Debug';
+
+var functions = require('../../bin/' + config + '/Content/Script/functions.js');
+
+var context = {};
+var logs = [];
+
+describe('functions', () => {
```
Comment (mathewc) [inline test/WebJobs.Script.Tests/TestScripts/Node/functions.tests.js]: Ah, I'll have to learn this describe/it this/it that test format. I'm used to the suite/test format :)

**Comment by mathewc** [comment]
:ship:

File: .gitignore [hunk 3]
```diff
@@ -14,3 +14,4 @@ Publish
 /TestResults
 
 /tools/NuGet.exe
+/node_modules
```
Comment (mamaso) [inline .gitignore]: msbuild WebJobs.Script.proj /t:MochaTest

I'll add a script to do it though

File: WebJobs.Script.proj (hunk 1 above)
Comment (fabiocav) [inline WebJobs.Script.proj]: Looks like that's already the case but I'm not sure how (if) AppVeyor sorts the tests.

**Pull request merged**